[2201.9.0] Add build tool documentation (#8835)
ballerina-platform/ballerina-dev-website: swan-lake/development-tutorials/source-code-dependencies/package-references.md

  scope =  "<scope-of-the-jar-file>"
  ```

### Tools

> You can specify code generation tools to integrate with the package build. These tools execute before the package build and generate code used during the build process.

Reviewer: ```suggestion
You can specify code generation tools to integrate with the package build. These tools execute before the package build and generate code that is essential for the build process.
```

Author: Fixed with 8ca99fb230b7749756c6b0f36bdd01f05cef3708

The following example shows how to specify a tool in the `Ballerina.toml` file.

```toml
[[tool.<command>]]
id = "<tool-id>"

> filePath = "<specification-file>"

Reviewer: ```suggestion
filePath = "<schema-or-API-specification-file>"
```

Author: Fixed with 8ca99fb230b7749756c6b0f36bdd01f05cef3708

> targetModule = "<generated-code-destination-module>"

Reviewer: ```suggestion
targetModule = "<destination-module-to-generated-code->"
```

Author: Fixed with 8ca99fb230b7749756c6b0f36bdd01f05cef3708

options.<option1> = "<value1>"
options.<option2> = "<value2>"
```

The tool command that you need to use should be specified after the `tool.` prefix in the table array header.

> The mandatory `id` field specifies a unique identifier for the tool entry, as there can be multiple entries for the same tool command to generate code for different modules. The `id` must consist of alphanumeric characters and underscores only, but it must not begin or end with an underscore, and consecutive underscores are not permitted.

Reviewer: ```suggestion
The mandatory `id` field specifies a unique identifier for the tool entry, as a tool can utilize multiple schemas/API specifications files. The `id` must consist of alphanumeric characters and underscores only, and must not begin or end with an underscore. Consecutive underscores are also not permitted.
```

Author: Fixed with 8ca99fb230b7749756c6b0f36bdd01f05cef3708

The `filePath` field is mandatory, providing the path to the specification file that the tool uses to generate code.
 
The `targetModule` field specifies the module where the generated code should be placed. If this is not specified, it will default to the root module. This should be unique for each tool entry.
 

> The `options` fields are defined by the tool developer and used to pass additional parameters to the tool.

Reviewer: ```suggestion
The `options` fields can be used to pass additional parameters to the tool.
```

Author: Fixed with 8ca99fb230b7749756c6b0f36bdd01f05cef3708

> If a tool contains multiple subcommands, you can specify them as follows.

Reviewer: ```suggestion
If a tool provides multiple subcommands, you can specify them as follows.
```

Author: Fixed with 8ca99fb230b7749756c6b0f36bdd01f05cef3708
